### PR TITLE
Update to axe-core 3.3.2

### PR DESCRIPTION
**Pull request by karanbirsingh**
#### Description of changes

This PR consumes axe-core 3.3.2 and includes:
- updates to rule-to-WCAG mappings, including `scrollable-region-focusable` and `label-content-name-mismatch` as `BestPractice` so they aren't run in the standard set
- an update to our `restoreScroll` option type, which [worked internally](https://github.com/dequelabs/axe-core/commit/d55f3cda4598320f8ee496684a01f8b9e1598f8d?short_path=8be2f64#diff-8be2f642524623d5012c7b2cf4fa4eb9) but wasn't included in axe.d.ts [until now](https://github.com/dequelabs/axe-core/pull/1697/files)
- updated launchpad & guidance snapshot
- update to color guidance to include interactive link to improve keyboard accessibility

example new rule result:
role-img-alt screenshot

updated use-of-color guidance to include WCAG link:
use of color screenshot

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)
- [x] (UI changes only) Added screenshots/GIFs to description above
- [x] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/scanner/rule-to-links-mappings.ts [hunk 1]
```diff
@@ -17,6 +17,13 @@ const BestPractice: HyperlinkDefinition = {
 };
 
 export const ruleToLinkConfiguration: DictionaryStringTo<HyperlinkDefinition[]> = {
+    'aria-input-field-name': [link.WCAG_4_1_2],
+    'aria-toggle-field-name': [link.WCAG_4_1_2],
+    'avoid-inline-spacing': [link.WCAG_1_4_12],
+    'input-button-name': [link.WCAG_4_1_2],
+    'landmark-unique': [BestPractice],
+    'role-img-alt': [link.WCAG_1_1_1],
+    'scrollable-region-focusable': [BestPractice],
```
Comment (karanbirsingh) [inline src/scanner/rule-to-links-mappings.ts]: `scrollable-region-focusable` should actually be mapped to 2_1_1, but it seems like mapping to `BestPractice` is how we exclude rules from being run, regardless of their original designation from `axe`. Am I missing something?

Comment (dbjorge) [reply src/scanner/rule-to-links-mappings.ts]: You're correct. I'd recommend mapping to both for clarity (this will still exclude them), similar to how `landmark-complementary-is-top-level` is marked up below

File: src/scanner/rule-to-links-mappings.ts [hunk 2]
```diff
@@ -17,6 +17,13 @@ const BestPractice: HyperlinkDefinition = {
 };
 
 export const ruleToLinkConfiguration: DictionaryStringTo<HyperlinkDefinition[]> = {
```
Comment (karanbirsingh) [inline src/scanner/rule-to-links-mappings.ts]: would like to alphabetize these in a different PR

File: src/scanner/rule-to-links-mappings.ts (hunk 1 above)
Comment (karanbirsingh) [reply src/scanner/rule-to-links-mappings.ts]: Good idea, updated

Comment (ferBonnin) [reply src/scanner/rule-to-links-mappings.ts]: this rule had a false positive, we dont want to enable it until we consume the fix too

**Comment by karanbirsingh** [comment]
The `axe-core` change looks good in validation. Checks are failing because:
- `scrollable-region-focusable` fails in the remote build (need to introduce a scrollbar locally) for `adhoc/color/guidance`. We exclude this rule in production because of a false positive edge case [here](https://github.com/dequelabs/axe-core/issues/1701). We need to understand whether our failure is a false positive or not, and update our E2E test or E2E ruleset accordingly. It looks like the color guidance page has no links or interactive elements
- `e2e linux tests 1` shows `Hamburger menu Normal mode should have content matching snapshot` failing because [Chrome failed to start](https://dev.azure.com/ms/accessibility-insights-web/_build/results?buildId=37229&view=ms.vss-test-web.build-test-results-tab&runId=1725124&resultId=100015&paneView=debug)
- One of the mac tabstop tests fails [here](https://dev.azure.com/ms/accessibility-insights-web/_build/results?buildId=37229&view=ms.vss-test-web.build-test-results-tab&runId=1725096&resultId=100258&paneView=debug)
